src/components/common/ClassroomEditor.jsx

```diff
@@ -0,0 +1,240 @@
+              </ListItem>
+            }
+            {(!(props.selectedClassroom.description && props.selectedClassroom.description.length > 0)) ? null :
+              <ListItem>
+                <Label className="secondary">Description</Label>
+                <span>{props.selectedClassroom.description}</span>
+              </ListItem>
+            }
+            <ListItem>
+              <Label className="secondary">Join Link</Label>
+              <span>{joinURL}</span>
+            </ListItem>
+          </List>
+        </Box>
+
+        <Table className="manager-table">
+          <thead className="manager-table__headers">
+            <TableRow>
+              <th id="student-name" scope="col" className="headers__header">Student Name/Zooniverse ID</th>
+              <th id="assignment-intro" scope="col" className="headers__header">Introduction</th>
+              <th id="assignment-galaxy" scope="col" className="headers__header">Galaxy Zoo 101</th>
+              <th id="assignment-hubble" scope="col" className="headers__header">Hubble's Law</th>
+              <th id="student-remove" scope="col" className="headers__header">Remove Student</th>
+            </TableRow>
+          </thead>
+          
+          <tbody className="manager-table__body">
+            {props.assignmentsStatus === ASSIGNMENTS_STATUS.FETCHING && (
+              <TableRow className="manager-table__row-data">
+                <td colSpan="4"><Spinning /></td>
+              </TableRow>
+            )}
+
+            {props.assignmentsStatus !== ASSIGNMENTS_STATUS.FETCHING &&
+              students.map((student) => {
+              return (
+                <TableRow className="manager-table__row-data" key={`classroom-student-${student.id}`}>
+                  <td headers="student-name">
+                    {(student.zooniverseDisplayName && student.zooniverseDisplayName.length > 0)
+                      ? <span>{student.zooniverseDisplayName}</span>
+                      : <span className="secondary">{student.zooniverseLogin}</span> }
+                  </td>
+                  <td headers="assignment-intro">{Math.floor(Math.random() * 100)}%</td>
+                  <td headers="assignment-galaxy">{Math.floor(Math.random() * 100)}%</td>
+                  <td headers="assignment-hubble">{Math.floor(Math.random() * 100)}%</td>
+                  <td headers="student-remove">
+                    <Button
+                      className="manager-table__button--delete"
+                      type="button"
+                      onClick={props.removeStudentFromClassroom.bind(null, props.selectedClassroom.id, student.id)}
+                      icon={<CloseIcon size="small" />}
+                    ></Button>
+                  </td>
+                </TableRow>
+              );
+            })}
+          </tbody>
+
+        </Table>
+      </Box>
+    );
+  }
+  
+  exportGrades() {
+    if (!this.props.selectedClassroom) return null;
+    
+    //TODO
```

Comment:
This is a functional example of a 'Download CSV data' feature, but I'd highly recommend looking at the SuperDownloadButton to see the (weird, wonky) workaround specific to handling cross-browser compatibility. Name... Safari.

Reply:
Yeah, I think the SuperDownloadButton is the way to go long term.